### ISSUE TITLE
Soundtouch 520 gets state "unknown" when turned on

### ISSUE THREAD

**Issue by markaragnos81** [thlucas1/homeassistantcomponent_soundtouchplus#25]
## System Information

version | core-2024.12.4
-- | --
installation_type | Home Assistant OS
dev | false
hassio | true
docker | true
user | root
virtualenv | false
python_version | 3.13.0
os_name | Linux
os_version | 6.6.66-haos
arch | x86_64
timezone | Europe/Berlin
config_dir | /config

<details><summary>Home Assistant Community Store</summary>

GitHub API | ok
-- | --
GitHub Content | ok
GitHub Web | ok
HACS Data | ok
GitHub API Calls Remaining | 4990
Installed Version | 2.0.1
Stage | running
Available Repositories | 1499
Downloaded Repositories | 37

</details>

<details><summary>Home Assistant Cloud</summary>

logged_in | false
-- | --
can_reach_cert_server | ok
can_reach_cloud_auth | ok
can_reach_cloud | ok

</details>

<details><summary>Home Assistant Supervisor</summary>

host_os | Home Assistant OS 14.1
-- | --
update_channel | stable
supervisor_version | supervisor-2024.12.0
agent_version | 1.6.0
docker_version | 27.2.0
disk_total | 62.3 GB
disk_used | 19.8 GB
healthy | true
supported | true
host_connectivity | true
supervisor_connectivity | true
ntp_synchronized | true
virtualization | kvm
board | ova
supervisor_api | ok
version_api | ok
installed_addons | File editor (5.8.0), Samba share (12.3.2), evcc (0.131.12), Matter Server (6.6.1), Terminal & SSH (9.16.0), Mosquitto broker (6.4.1), Zigbee2MQTT (1.42.0-2), eufy-security-ws (1.9.1), go2rtc master (master)

</details>

<details><summary>Dashboards</summary>

dashboards | 5
-- | --
resources | 20
views | 44
mode | storage

</details>

<details><summary>Recorder</summary>

oldest_recorder_run | 10. Dezember 2024 um 15:01
-- | --
current_recorder_run | 20. Dezember 2024 um 08:32
estimated_db_size | 1997.73 MiB
database_engine | sqlite
database_version | 3.45.3

</details>

<details><summary>Bose SoundTouch Plus</summary>

integration_version | v1.0.75
-- | --
devices_configured | 3: Bose Wohnzimmer (SoundTouch 20), Bose SoundTouch 520 (Cinemate), Bose Portable (SoundTouch Portable)

</details>

### Checklist

- [X] I have enabled SmartInspect debug logging for my installation.
- [X] I have filled out the issue template to the best of my ability.
- [X] This issue only contains 1 issue (if you have multiple issues, open one issue for each issue).
- [X] This issue is not a duplicate issue of any [previous issues](https://github.com/thlucas1/homeassistantcomponent_soundtouchplus/issues?q=is%3Aissue+label%3A%22Bug%22+)..

### Describe the issue

Actually, it's just as simple as the steps to reproduce it. If the SoundTouch 520 is turned off, the status is correctly set to 'off'. When you turn on the SoundTouch 520, the status is 'unknown'. Additionally, I am using a SoundTouch 20 (Gen 2) and a SoundTouch Portable (Gen 2). Everything works correctly with them.

### Reproduction steps

1. Turn on: State = Unknown
2. Turn off: State = off
3. Turn on: State = Unknown
4. Turn off: State = off

### Debug logs

```text
21. Dezember 2024
Bose SoundTouch 520 wurde unbekannt
12:55:40 - Vor 23 Minuten
20. Dezember 2024
Bose SoundTouch 520 ausgeschaltet
23:08:27 - Vor 14 Stunden
Bose SoundTouch 520 wurde unbekannt
20:28:23 - Vor 17 Stunden
Bose SoundTouch 520 ausgeschaltet
18:20:30 - Vor 19 Stunden
Bose SoundTouch 520 wurde unbekannt
17:23:32 - Vor 20 Stunden
Bose SoundTouch 520 ausgeschaltet
16:57:22 - Vor 20 Stunden
Bose SoundTouch 520 wurde unbekannt
13:00:23 - Gestern
```

### Diagnostics dump

not providet from integration

**Comment by thlucas1** [comment]
Not for sure on how the state would read on a [Bose SoundTouch 520](https://github.com/thlucas1/homeassistantcomponent_soundtouchplus/wiki/SoundTouch-Model-Notes#soundtouch-520), as it utilizes the [SoundTouch Wireless Adapter](https://github.com/thlucas1/homeassistantcomponent_soundtouchplus/wiki/SoundTouch-Model-Notes#soundtouch-wireless-adapter).  I don't own any of those, and have no way to test it.  

Do you know if the SoundTouch Wireless Adapter is a verion 1? or version 2 (model number ends with “SM2”)?  I know the version 2 adds Bluetooth and gives more options within HA for controlling.

**Comment by markaragnos81** [comment]
You are absolutely right. It is a SoundTouch 520 where a wireless adapter is being used. I checked my wireless adapter and can confirm that I am using version 2.

**Comment by thlucas1** [comment]
After turning the 520 to on, issue the following request from a desktop browser window (change ip address to your 520’s IP address):

http://192.168.1.131:8090/nowPlaying

What are the results?

**Comment by markaragnos81** [comment]
I have 2 different scenarios:

**1. Input: SoundTouch**
<nowPlaying deviceID="EC24B8BD894F" source="TUNEIN" sourceAccount="">
<ContentItem source="TUNEIN" type="stationurl" location="/v1/playback/station/s267604" sourceAccount="" isPresetable="true">
<itemName>RADIO BOB! Rock Hits</itemName>
<containerArt>http://cdn-profiles.tunein.com/s267604/images/logog.jpg?t=637619812720000000</containerArt>
</ContentItem>
<track>RADIO BOB! Rock Hits</track>
<artist>Die größten Hits der angesagtesten Rockbands – nonstop!</artist>
<album/>
<stationName>RADIO BOB! Rock Hits</stationName>
<art artImageStatus="IMAGE_PRESENT">http://cdn-profiles.tunein.com/s267604/images/logog.jpg?t=637619812720000000</art>
<favoriteEnabled/>
<playStatus>PLAY_STATE</playStatus>
<streamType>RADIO_STREAMING</streamType>
</nowPlaying>

**2. Input: TV**
<nowPlaying deviceID="EC24B8BD894F" source="LOCAL">
<ContentItem source="LOCAL" isPresetable="true"/>
</nowPlaying>

In scenario 1, everything is displayed correctly. In scenario 2, it is the situation I described in the bug.

**Comment by thlucas1** [comment]
That makes sense, as the SOURCE="LOCAL"; this is a known limitation of the Wireless Adapter (both version1 and version2).

This is not considered a bug, as the 520 does not report any detailed nowplaying information when the SOURCE=LOCAL (hence the `unknown` status).  It's state is not `off`, but is `unknown` (e.g. it could be playing, paused, etc, but we don't know because it can't tell us that information).

Per [other user's notes](https://github.com/thlucas1/homeassistantcomponent_soundtouchplus/wiki/SoundTouch-Model-Notes#user-notes) (that own the 520), the local source selection cannot be controlled or monitored.
